plugin/magit.vim

```diff
@@ -293,7 +293,13 @@ endfunction
 function! s:mg_get_commit_msg(...)
 	let commit_section_pat_start='^'.g:magit_sections.commit.'$'
 	" Get next section pattern with g:magit_default_sections order
-	let commit_section_pat_end='^'.g:magit_sections[g:magit_default_sections[match(g:magit_default_sections, 'commit')+1]].'$'
+	if ( match(g:magit_default_sections, 'commit')+1 >= len(g:magit_default_sections) )
```

Comment:
use intermediate variable to store `match(g:magit_default_sections, 'commit')` result
check if this position is equal to len + 1